src/Build.UnitTests/BuildEventArgsSerialization_Tests.cs

```diff
@@ -848,10 +849,246 @@ public void ReadingCorruptedStreamThrows()
                 var binaryReader = new BinaryReader(memoryStream);
                 using var buildEventArgsReader = new BuildEventArgsReader(binaryReader, BinaryLogger.FileFormatVersion);
 
-                Assert.Throws<EndOfStreamException>(() => buildEventArgsReader.Read());
+                try
+                {
+                    buildEventArgsReader.Read();
+                }
+                catch (Exception e)
```

Comment:
For fidelity with the prior version, you also need to assert that an exception was caught; I'd prefer an `Assert.ThrowsAny<Exception>()` + type checks.

Reply:
Reverting my change.

Made the `EndOfStream` detection more granular - so the 'real' end of stream continues to be thrown unwrapped.